spec/requests/people_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+describe '/people', type: :request do
+  describe 'GET /index' do
+    let!(:people) { create_list(:person, 40) }
+
+    before do
+      sign_in create(:user)
+    end
+
+    it 'renders a successful response' do
+      get people_url
+      expect(response).to be_successful
+    end
+
+    it 'paginates results' do
+      get people_url
+      expected, remainder = people.partition.with_index { |p, i| i < Pagy::VARS[:items] }
+
+      expected.each do |person|
+        expect(response.body).to match(/#{person.name}/)
+      end
+
+      remainder.each do |person|
+        expect(response.body).to_not match(/#{person.name}/)
+      end
```

Comment:
I played around with a variation that counts the number of people:
```suggestion
      people_links = response.body.scan %r{href="/people/\d+/.+"}
      expect(people_links.size).to eq Pagy::VARS[:items]
```
But it didn't really speed things up much (both are coming in ~1.9s).

I think the people creation actually takes up a chunk of that time (~.5s), which i imagine would be incurred with a controller test as well.

This is probably part of the overhead with request specs :(

Reply:
In that case, do you think leaving it like it is would be fine?